src/Planners/include/BipedalLocomotion/Planners/DCMPlanner.h

```diff
@@ -0,0 +1,88 @@
+/**
+ * @file DCMPlanner.h
+ * @authors Giulio Romualdi
+ * @copyright 2020 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#ifndef BIPEDAL_LOCOMOTION_PLANNERS_DCM_PLANNER_H
+#define BIPEDAL_LOCOMOTION_PLANNERS_DCM_PLANNER_H
+
+#include <memory>
+
+#include <Eigen/Dense>
+
+#include <BipedalLocomotion/ParametersHandler/IParametersHandler.h>
+#include <BipedalLocomotion/Planners/ContactPhaseList.h>
+#include <BipedalLocomotion/System/Advanceable.h>
+
+namespace BipedalLocomotion
+{
+namespace Planners
+{
+
+/**
+ * DCMPlannerState describes the state of the Divergent Component of Motion (DCM) planner.
+ */
+struct DCMPlannerState
+{
+    Eigen::Vector3d dcmPosition; /**< Position of the DCM expressed w.r.t. the inertial frame */
+    Eigen::Vector3d dcmVelocity; /**< Velocity of the DCM expressed w.r.t. the inertial frame */
+    Eigen::Vector3d vrpPosition; /**< Position of the virtual repellent point (VRP) expressed
+                                    w.r.t. the inertial frame */
+    double omega;/**< Value of the parameter omega */
+};
+
+/**
+ * DCMPlanner defines a trajectory generator for the variable height Divergent
+ * component of motion (DCM). Please inherit publicly from this class in order to define your
+ * planner.
+ */
+class DCMPlanner : public BipedalLocomotion::System::Advanceable<DCMPlannerState>
+{
+protected:
+    std::shared_ptr<const ContactPhaseList> m_contactPhaseList; /**< Pointer containing the contact
+                                                                   phases. */
+
+    DCMPlannerState m_initialState; /**< Initial state of the planner */
+
+public:
+    /**
+     * Initialize the planner.
+     * @param handler pointer to the parameter handler.
+     * @return true in case of success/false otherwise.
+     */
+    virtual bool initialize(std::weak_ptr<ParametersHandler::IParametersHandler> handler);
```

Comment:
I think it may make sense to pass here directly the shared pointer (the same below).

Reply:
Done in https://github.com/dic-iit/bipedal-locomotion-framework/pull/61/commits/6fb1de6774b16a94415c4a9daa0f692ed63840ce. 

When you say `the same below` are you referring to `DCMPlanner::setContactPhaseList()`?

Reply:
> When you say `the same below` are you referring to [`DCMPlanner::setContactPhaseList()`](https://github.com/dic-iit/bipedal-locomotion-framework/blob/6fb1de6774b16a94415c4a9daa0f692ed63840ce/src/Planners/include/BipedalLocomotion/Planners/DCMPlanner.h#L71)?

Yes

Reply:
Done in 7aaa1cb0786a9b01dcd3fcf9fc27b22d32a95bb2